src/web3/KeychainSigner.ts

```diff
@@ -239,58 +59,26 @@ export class KeychainSigner implements Signer {
     if (!privateKey) {
       return false
     }
-
-    this.unlockedLocalSigner = new LocalSigner(privateKey)
-    this.unlockTime = Date.now()
-    this.unlockDuration = duration
-    return true
-  }
-
-  isUnlocked(): boolean {
-    if (this.unlockDuration === undefined || this.unlockTime === undefined) {
-      return false
-    }
-
-    if (this.unlockDuration === 0) {
-      return true
-    }
-
-    return this.unlockTime + this.unlockDuration * 1000 > Date.now()
-  }
-
-  /**
-   * Updates the passphrase of an account
-   * @param oldPassphrase - the passphrase currently associated with the account
-   * @param newPassphrase - the new passphrase to use with the account
-   * @returns whether the update was successful
-   */
-  async updatePassphrase(oldPassphrase: string, newPassphrase: string) {
-    const privateKey = await getStoredPrivateKey(this.account, oldPassphrase)
-    if (!privateKey) {
-      return false
-    }
-    await storePrivateKey(privateKey, this.account, newPassphrase)
+    this.localSigner = new LocalSigner(privateKey)
+    await this.lock.unlock(this.account.address, passphrase, duration)
     return true
   }
 
   async decrypt(ciphertext: Buffer): Promise<Buffer> {
-    return this.localSigner.decrypt(ciphertext)
+    return this.unlockedLocalSigner.decrypt(ciphertext)
   }
 
   async computeSharedSecret(publicKey: string): Promise<Buffer> {
-    return this.localSigner.computeSharedSecret(publicKey)
+    return this.unlockedLocalSigner.computeSharedSecret(publicKey)
   }
 
-  /**
-   * Get the local signer. Throws if not unlocked.
-   */
-  protected get localSigner(): LocalSigner {
-    if (!this.isUnlocked()) {
-      this.unlockedLocalSigner = null
-    }
-    if (!this.unlockedLocalSigner) {
+  protected get unlockedLocalSigner(): LocalSigner {
```

Comment:
So we've flipped localSigner and unlockedLocalSigner? And should we keep the existing functionality of setting the localSigner instance variable to null if not unlocked? 
Also can we add some unit tests for at least this and the unlock method?

Reply:
I flipped back the name, the switch was leftover from a previous iteration. KeychainWallet covers these methods pretty well so I don't think they are necessary to test.